light-node/src/commands/initialize.rs

```diff
@@ -0,0 +1,92 @@
+//! `intialize` subcommand
+
+use crate::application::app_config;
+use crate::config::LightClientConfig;
+
+use abscissa_core::{Command, Options, Runnable};
+use std::collections::HashMap;
+
+use tendermint::lite::Header;
+use tendermint::{hash, Hash};
+
+use tendermint_light_client::components::io::{AtHeight, Io, ProdIo};
+use tendermint_light_client::operations::ProdHasher;
+use tendermint_light_client::predicates::{ProdPredicates, VerificationPredicates};
+use tendermint_light_client::store::sled::SledStore;
+use tendermint_light_client::store::LightStore;
+use tendermint_light_client::types::Status;
+
+/// `intialize` subcommand
+#[derive(Command, Debug, Default, Options)]
+pub struct InitCmd {
+    #[options(
+        free,
+        help = "subjective height of the initial trusted state to initialize the node with"
+    )]
+    pub height: u64,
+
+    #[options(
+        free,
+        help = "hash of the initial subjectively trusted header to initialize the node with"
+    )]
+    pub header_hash: String,
+}
+
+impl Runnable for InitCmd {
+    fn run(&self) {
+        let subjective_header_hash =
+            Hash::from_hex_upper(hash::Algorithm::Sha256, &self.header_hash).unwrap();
+        let app_cfg = app_config();
+
+        let lc = app_cfg.light_clients.first().unwrap();
+
+        let mut peer_map = HashMap::new();
+        peer_map.insert(lc.peer_id, lc.address.clone());
+
+        let io = ProdIo::new(peer_map, Some(app_cfg.rpc_config.request_timeout));
+
+        initialize_subjectively(self.height, subjective_header_hash, &lc, &io);
+    }
+}
+
+fn initialize_subjectively(
```

Comment:
This function does too many things I believe, the database setup seems in the right place but the actual logic smells of library code.

Reply:
As it is something we potentially wanna test.

Reply:
I agree, a version of this should potentially live in the light-client crate (with tests)! 
Do you want to help out here? Otherwise I think it makes sense to capture this in an issue (actually #293 is related) before this gets merged and address it in a followup PR to not block this one.

Reply:
Capturing these things in a follow-up sounds great.